pkg/newlog/cmd/newlogd.go

```diff
@@ -995,41 +1025,48 @@ func checkKeepQuota() {
 	maxSize := int64(limitGzipFilesMbyts * 1000000)
 	sfiles := make(map[string]gfileStats)
 
-	key0, size0, err := checkDirGZFiles(sfiles, keepSentDir)
+	filesKeepSent, sizeKeepSent, err := checkDirGZFiles(sfiles, keepSentDir)
 	if err != nil {
 		log.Errorf("checkKeepQuota: keepSentDir %v", err)
 	}
-	key1, size1, err := checkDirGZFiles(sfiles, uploadAppDir)
+	filesAppUpload, sizeAppUpload, err := checkDirGZFiles(sfiles, uploadAppDir)
 	if err != nil {
 		log.Errorf("checkKeepQuota: AppDir %v", err)
 	}
-	key2, size2, err := checkDirGZFiles(sfiles, uploadDevDir)
+	filesDevUpload, sizeDevUpload, err := checkDirGZFiles(sfiles, uploadDevDir)
 	if err != nil {
 		log.Errorf("checkKeepQuota: DevDir %v", err)
 	}
-	key3, size3, err := checkDirGZFiles(sfiles, failSendDir)
+	fileFailSend, sizeFailSend, err := checkDirGZFiles(sfiles, failSendDir)
 	if err != nil && !os.IsNotExist(err) {
 		log.Errorf("checkKeepQuota: FailToSendDir %v", err)
 	}
 
-	totalsize := size0 + size1 + size2 + size3
-	totalCount := len(key0) + len(key1) + len(key2) + len(key3)
+	totalsize := sizeKeepSent + sizeAppUpload + sizeDevUpload + sizeFailSend
+	totalCount := len(filesKeepSent) + len(filesAppUpload) + len(filesDevUpload) + len(fileFailSend)
 	removed := 0
 	// limit file count to not as they can have less size than expected
 	// we can have enormous number of files
 	maxCount := int(maxSize / maxGzipFileSize)
 	if totalsize > maxSize || totalCount > maxCount {
-		keys := key0
-		keys = append(keys, key1...)
-		keys = append(keys, key2...)
-		keys = append(keys, key3...)
-		sort.Strings(keys)
-
-		for _, key := range keys {
-			if _, ok := sfiles[key]; !ok {
+		allFiles := filesKeepSent
+		allFiles = append(allFiles, filesAppUpload...)
+		allFiles = append(allFiles, filesDevUpload...)
+		allFiles = append(allFiles, fileFailSend...)
+
+		// please notice that the files will be removed in alphabetical order,
+		// starting from the oldest (lowest) timestamp
+		// app.log...gz
+		// dev.log.keep...gz
+		// dev.log.upload....gz
+		// TODO: declare a clear order of removal by directory
```

Comment:
what is format for saved files? It's not always that alphabetical <=> chronological, i.e. if we don't put zero in front of day/month, it won't work, f.e. '9' > '11' since we do symbol-wise comparison

Reply:
the files are named like this:
app.6656f860-7563-4bbf-8bba-051f5942982b.log.1730464687367.gz
dev.log.keep.1730404601953.gz
dev.log.upload.1730404601953.gz

so we use timestamps instead of dates and the alphabetical order should also represent the time order.
However I understand that alphabetical order is not ideal and that's why I put that todo for myself :)